cmd/object-api-utils.go

```diff
@@ -597,3 +598,39 @@ func (g *GetObjectReader) Read(p []byte) (n int, err error) {
 	}
 	return
 }
+
+// PutObjectReader is a type that wraps sio.EncryptReader and
+// underlying hash.Reader in a struct
+type PutObjectReader struct {
+	DataReader *hash.Reader // actual data stream
+	OrigReader *hash.Reader // original data stream
```

Comment:
The two readers above are named a bit vaguely? The names need to disambiguate the semantics better.

Reply:
Maybe:
```
Plaintext *hash.Reader
Ciphertext *hash.Reader
```
?
Making it quite obvious for non-encrypted requests, too...